lib/dash/dash_parser.js

```diff
@@ -1444,11 +1457,35 @@ shaka.dash.DashParser = class {
         node: elem,
         isLastPeriod: periodDuration == null || !next,
       };
-      const period = this.parsePeriod_(context, getBaseUris, info);
+      let period = null;
+      let signature = null;
+      if (enablePeriodCaching) {
+        const cacheKey = this.getPeriodCacheKey_(periodId, start);
+        const cachedEntry = (!newPeriod && this.manifest_) ?
+            this.periodCache_.get(cacheKey) : null;
+        let reusePeriod = false;
+        if (cachedEntry) {
+          signature = this.computePeriodSignature_(elem);
```

Comment:
It's unclear to me why we need both a period cache key and a separate signature...